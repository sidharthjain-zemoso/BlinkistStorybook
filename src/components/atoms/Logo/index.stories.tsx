import { ComponentStory, ComponentMeta } from "@storybook/react";

import Logo from "./index";

export default {
  title: "Atoms/Logo",
  component: Logo,
} as ComponentMeta<typeof Logo>;

const Template: ComponentStory<typeof Logo> = (args) => <Logo {...args} />;

export const BlinkistLogo: ComponentStory<typeof Logo> = Template.bind({});
BlinkistLogo.args = {
  url: "/res/logo/Blinklist.png",
  sx: {},
};
