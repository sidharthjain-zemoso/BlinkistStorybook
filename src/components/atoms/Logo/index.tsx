type LogoProps = {
  id?: string;
  url: string;
  sx?: Object;
};

const Logo: React.FC<LogoProps> = (props) => {
  return (
    <img
      data-testid={props.id}
      id={props.id}
      src={process.env.PUBLIC_URL + props.url}
      style={props.sx}
      alt="Logo"
    />
  );
};

export default Logo;
