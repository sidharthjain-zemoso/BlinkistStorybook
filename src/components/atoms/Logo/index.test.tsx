import { render, screen } from "@testing-library/react";
import Logo from "./";

test("renders icon component", () => {
  render(<Logo url="res/logo/Blinklist.png" />);

  const imageElement = screen.getByRole("img");
  expect(imageElement).toMatchSnapshot();
});
