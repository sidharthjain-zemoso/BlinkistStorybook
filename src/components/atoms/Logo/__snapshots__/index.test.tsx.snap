// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders icon component 1`] = `
<img
  alt="Logo"
  src="res/logo/Blinklist.png"
/>
`;
